src/sentry/integrations/source_code_management/metrics.py

```diff
@@ -56,7 +59,22 @@ def get_interaction_type(self) -> str:
         return str(self.interaction_type)
 
     def get_extras(self) -> Mapping[str, Any]:
+        def get_id(obj: Any) -> int | None:
+            if hasattr(obj, "id"):
+                return obj.id
+            if isinstance(obj, int):
+                return obj
+            return None
```

Comment:
This more complex type handling is a code smell. Are we maybe trying to pack too much into the metrics provider by requiring all of these fields up front? Should the caller be in charge of providing the extra context itself via the lifecycle object since there may or may not be an org, installation, etc?